### ISSUE TITLE
After installing 2.1.300, dotnet new no longer creates instances of templates

### ISSUE THREAD

**Issue by livarcocc** [dotnet/templating#1584]
_From @simmdan on June 15, 2018 6:31_

## Steps to reproduce
dotnet new console (or any other template name)

## Expected  behavior
New console project created.

## Actual behavior
No error, but no files created.  Command just returns to the console with no message.
Executable exit code is: 2097158

## Environment data
`dotnet --info` output:

.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300\

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.300 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

## Other Info:
One of my computers works fine.  This one, however, does not.  I have tried uninstalling and reinstalling the dotnet sdk.  Rebooting the computer.  Uninstalling all other versions of the dotnet sdk, etc.  Same results.

_Copied from original issue: dotnet/cli#9499_

**Comment by mlorbetske** [comment]
That return code is "Not Found", is there any output at all from the command or just an immediate return to the prompt?

**Comment by mlorbetske** [comment]
Could you post the ouptuts of `dotnet new --list` and one of the command lines you're seeing this behavior with please?

**Comment by mlorbetske** [comment]
If no templates are displayed in `dotnet new --list`, try running `dotnet new --debug:reinit` to get it to redo its initialization

**Comment by mlorbetske** [comment]
It looks like for this to happen, the template would have to be included in the cache (to be able to make it to `InstantiateAsync`), but failed to load. https://github.com/dotnet/templating/blob/stabilize/src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs#L62-L68

`New3Command.CreateTemplateAsync` doesn't have specific handling for `CreationResultStatus.NotFound`, so it falls through the default case, skipping printing a message. We should fix that, but we'll also need to understand what are the possible causes in order to give a good message about what to do next.

**Comment by simmdan** [comment]
Thanks for the responses!  I don't have access to the machine with this issue until I get home from work tonight.  I'll try these things and report back then.

**Comment by simmdan** [comment]
```
C:\tmp\foo> dotnet new --list
Usage: new [options]

Options:
  -h, --help          Displays help for this command.
  -l, --list          Lists templates containing the specified name. If no name is specified, lists all templates.
  -n, --name          The name for the output being created. If no name is specified, the name of the current directory is used.
  -o, --output        Location to place the generated output.
  -i, --install       Installs a source or a template pack.
  -u, --uninstall     Uninstalls a source or a template pack.
  --nuget-source      Specifies a NuGet source to use during install.
  --type              Filters templates based on available types. Predefined values are "project", "item" or "other".
  --force             Forces content to be generated even if it would change existing files.
  -lang, --language   Filters templates based on language and specifies the language of the template to create.


Templates                                         Short Name         Language          Tags
----------------------------------------------------------------------------------------------------------------------------
Console Application                               console            [C#], F#, VB      Common/Console
Class library                                     classlib           [C#], F#, VB      Common/Library
Unit Test Project                                 mstest             [C#], F#, VB      Test/MSTest
xUnit Test Project                                xunit              [C#], F#, VB      Test/xUnit
Razor Page                                        page               [C#]              Web/ASP.NET
MVC ViewImports                                   viewimports        [C#]              Web/ASP.NET
MVC ViewStart                                     viewstart          [C#]              Web/ASP.NET
ASP.NET Core Empty                                web                [C#], F#          Web/Empty
ASP.NET Core Web App (Model-View-Controller)      mvc                [C#], F#          Web/MVC
ASP.NET Core Web App                              razor              [C#]              Web/MVC/Razor Pages
ASP.NET Core with Angular                         angular            [C#]              Web/MVC/SPA
ASP.NET Core with React.js                        react              [C#]              Web/MVC/SPA
ASP.NET Core with React.js and Redux              reactredux         [C#]              Web/MVC/SPA
Razor Class Library                               razorclasslib      [C#]              Web/Razor/Library/Razor Class Library
ASP.NET Core Web API                              webapi             [C#], F#          Web/WebAPI
global.json file                                  globaljson                           Config
NuGet Config                                      nugetconfig                          Config
Web Config                                        webconfig                            Config
Solution File                                     sln                                  Solution

**Comment by simmdan** [comment]
If I run `dotnet new some-template-that-doesnotexist`, I get a response that includes the message `No templates matched the input template name: some-template-that-doesnotexist.`

But if I do `dotnet new console` or `dotnet new sln` (or presumably any other template that is in the list), I just get my console prompt back with no output and no files created.

I just tried `dotnet new --debug:reinit`, and that seems to have fixed my problem.  Now I can run `dotnet new console` and it creates things as expected.  That's great for me, but unfortunately it means I just destroyed the repro of this issue.  Sorry.

**Comment by mlorbetske** [comment]
Glad you're unblocked. In that case I think we may just add generic messaging for that case suggesting re-initializing. We've been trying to avoid public documentation of that flag as it's a big hammer to use, but in this case it might be the only option.

The only thing that came to mind when trying to figure out how to get in to this state - did you do like a ZIP install to try out the CLI first, delete it and then install it? When `new` initializes, it only takes in to account the id/version of the caller (in this case `dotnet new` and `2.1.300` and builds up a cache of the templates that are available and their locations on disk - if those locations are no longer valid, but the cache is still present (it's stored in the user profile directory, so wouldn't travel with the CLI if the location changed), it could cause this.

**Comment by simmdan** [comment]
No, I did not install by zip--I believe I used the chocolatey dotnetcore-sdk package to install.  That said I did have some weird behavior on my machine (not sure if it was just this issue that I didn't understand or was something else), so I uninstalled all of the .net core SDKs and uninstalled VS 2017 and then reinstalled things.  I wonder if there might have been something strange about the interleaving order of those operations that caused it.  Like maybe uninstalling .net core out from under vs and then uninstalling and reinstalling vs and then reinstalling the cli/sdk or something...  Hard to say.
